docs/source/app_developers_guide/docker.rst

```diff
@@ -490,3 +461,234 @@ look similar to this:
+terminal:
+
+.. code-block:: console
+
+  $ docker exec -it sawtooth-tp_intkey_python-default bash
+
+To see which components are running, run this command from the container:
+
+.. code-block:: console
+
+  $ ps -ef | grep python | grep -v grep
+  root         1     0  0 21:20 ?        00:00:00 /usr/bin/python3 /usr/bin/tp_intkey_python -vv tcp://validator:4004
+
+
+The XO Transaction Processor Container
+--------------------------------------
+
+* Runs a single XO transaction processor
+* Hostname: tp_xo_python
+* Handles transactions of the *XO transaction family*
+
+Log into this container by running this command from the host computer's
+terminal:
+
+.. code-block:: console
+
+  $ docker exec -it sawtooth-tp_xo_python-default bash
+
+To see which components are running, run this command from the container:
+
+.. code-block:: console
+
+  $ ps -ef | grep python | grep -v grep
+  root         1     0  0 21:20 ?        00:00:00 /usr/bin/python3 /usr/bin/tp_xo_python -vv tcp://validator:4004
+
+
+Viewing Log Files
+=================
+
+You can view the log files for any running Docker container using the
+following command:
+
+.. code-block:: console
+
+  $ docker logs CONTAINER
+
+
+.. note:: 
+
+  Replace CONTAINER with the name of one of the Sawtooth Docker containers,
+  e.g. ``sawtooth-validator-default``.
+
+
+Settings Transaction Family Usage
+=================================
+
+Sawtooth provides a :doc:`settings transaction family
+<../transaction_family_specifications/settings_transaction_family>` that stores
+on-chain configuration settings, along with a Settings family transaction
+processor written in Python.
+
+One of the on-chain settings is the list of supported transaction families.
+In the example below, a JSON array is submitted to the ``sawtooth config``
+command, which creates and submits a batch of transactions containing the
+configuration change.
+
+The submitted JSON array tells the validator or validator network to accept
+transactions of the following types:
+
+* intkey
+* sawtooth_settings
+
+To create and submit the batch containing the new setting, enter the following
+commands:
+
+.. note::
+
+  The config command needs to use a key generated in the validator container.
+  Thus, you must open a terminal window running in the validator container,
+  rather than the client container (for the following command only).
+  Run the following command from your host machine's CLI:
+
+.. code-block:: console
+
+  % docker exec -it sawtooth-validator-default bash
+
+Then run the following commands from the validator container:
+
+.. code-block:: console
+
+  $ sawtooth config proposal create \
+    --url http://rest_api:8080 \
+    --key /root/.sawtooth/keys/my_key.priv \
+    sawtooth.validator.transaction_families='[{"family": "intkey", "version": "1.0", "encoding": "application/protobuf"}, {"family":"sawtooth_settings", "version":"1.0", "encoding":"application/protobuf"}]'
```

Comment:
This probably needs to be fixed in a few places in the docs. This caught somebody with a very strange error in rocketchat over the weekend. The log message from executor is "failing transaction %s of type(name=intkey, ...) since it isn't required in the configuration.